src/main/host/syscall/handler/signal.rs

```diff
@@ -1,20 +1,82 @@
+use linux_api::errno::Errno;
+use linux_api::signal::{siginfo_t, Signal};
 use shadow_shim_helper_rs::syscall_types::ForeignPtr;
 use syscall_logger::log_syscall;
 
 use crate::cshadow as c;
-use crate::host::syscall::handler::{SyscallContext, SyscallHandler};
+use crate::host::process::Process;
+use crate::host::syscall::handler::{SyscallContext, SyscallHandler, ThreadContext};
 use crate::host::syscall::types::SyscallError;
 
 impl SyscallHandler {
     #[log_syscall(/* rv */ std::ffi::c_int, /* pid */ linux_api::posix_types::kernel_pid_t,
                   /* sig */ std::ffi::c_int)]
     pub fn kill(
         ctx: &mut SyscallContext,
-        _pid: linux_api::posix_types::kernel_pid_t,
-        _sig: std::ffi::c_int,
+        pid: linux_api::posix_types::kernel_pid_t,
+        sig: std::ffi::c_int,
     ) -> Result<(), SyscallError> {
-        let rv = Self::legacy_syscall(c::syscallhandler_kill, ctx)?;
-        assert_eq!(0, i32::from(rv));
+        log::trace!("kill called on pid {pid} with signal {sig}");
+
+        let pid = if pid == -1 {
+            // kill(2): If pid equals -1, then sig is sent to every process for which the calling
+            // process has permission to send signals, except for process 1.
+            //
+            // Currently unimplemented, and unlikely to be needed in the context of a shadow
+            // simulation.
+            log::warn!("kill with pid=-1 unimplemented");
+            return Err(Errno::ENOTSUP.into());
+        } else if pid == 0 {
+            // kill(2): If pid equals 0, then sig is sent to every process in the process group of
+            // the calling process.
+            //
+            // Currently every emulated process is in its own process group.
```

Comment:
Oops, this comment is wrong. I just filed https://github.com/shadow/shadow/issues/3315; could you update the comment and link to that?

Reply:
Added.